NorthstarDLL/engine/r2engine.h

```diff
@@ -202,63 +211,56 @@ namespace R2
-			// 
-			//   - While receiving network packets (like in PreDataUpdate/PostDataUpdate and proxies),
-			//     this is set to the SERVER TICKCOUNT for that packet. There is no interval between
-			//     the server ticks.
-			//     [server_current_Tick * tick_interval]
-			//
-			//   - While rendering, this is the exact client clock 
-			//     [client_current_tick * tick_interval + interpolation_amount]
-			//
-			//   - During prediction, this is based on the client's current tick:
-			//     [client_current_tick * tick_interval]
-			float m_flCurTime;
-		)
-
-		FIELDS(0x30,
-			// Time spent on last server or client frame (has nothing to do with think intervals)
-			float m_flFrameTime;
-
-			// current maxplayers setting
-			int m_nMaxClients;
-		)
-
-		FIELDS(0x3C,
-			// Simulation ticks - does not increase when game is paused
-			uint32_t m_nTickCount; // this is weird and doesn't seem to increase once per frame?
-
-			// Simulation tick interval
-			float m_flTickInterval;
-		)
-
-		FIELDS(0x60,
-			const char* m_pMapName;
-			int m_nMapVersion;
-		)
-
-		//FIELD(0x98, double m_flRealTime); // again?
-	};
-	// clang-format on
+		// Absolute time (per frame still - Use Plat_FloatTime() for a high precision real time
+		//  perf clock, but not that it doesn't obey host_timescale/host_framerate)
+		double m_flRealTime; // 0x0 ( Size: 8 )
+
+		// Absolute frame counter - continues to increase even if game is paused
+		int m_nFrameCount; // 0x8 ( Size: 4 )
+
+		// Non-paused frametime
+		float m_flAbsoluteFrameTime; // 0xc ( Size: 4 )
+
+		// Current time
+		//
+		// On the client, this (along with tickcount) takes a different meaning based on what
+		// piece of code you're in:
+		//
+		//   - While receiving network packets (like in PreDataUpdate/PostDataUpdate and proxies),
+		//     this is set to the SERVER TICKCOUNT for that packet. There is no interval between
+		//     the server ticks.
+		//     [server_current_Tick * tick_interval]
+		//
+		//   - While rendering, this is the exact client clock
+		//     [client_current_tick * tick_interval + interpolation_amount]
+		//
+		//   - During prediction, this is based on the client's current tick:
+		//     [client_current_tick * tick_interval]
+		float m_flCurTime; // 0x10 ( Size: 4 )
+		char _unk1[28]; // 0x14 ( Size: 28 )
+
+		// Time spent on last server or client frame (has nothing to do with think intervals)
+		float m_flFrameTime; // 0x30 ( Size: 4 )
+
+		// current maxplayers setting
+		int m_nMaxClients; // 0x34 ( Size: 4 )
+		char _unk2[4]; // 0x38 ( Size: 4 )
+
+		// Simulation ticks - does not increase when game is paused
+		//   this is weird and doesn't seem to increase once per frame?
+		uint32_t m_nTickCount; // 0x3c ( Size: 4 )
+
+		// Simulation tick interval
+		float m_flTickInterval; // 0x40 ( Size: 4 )
+		char _unk3[28]; // 0x44 ( Size: 28 )
+
+		const char* m_pMapName; // 0x60 ( Size: 8 )
+		int m_nMapVersion; // 0x68 ( Size: 4 )
+	} CGlobalVars;
```

Comment:
`m_pMapName` and `m_nMapVersion` are in `CGlobalVars` ( second link ). `m_nTickCount` is missing because i didn't confirm its location. Regarding classes: they are the same as structs but members are private by default. If we want to mimic source we should mimic classes/structs as much as we can even though they're the same in cpp.

Reply:
> Regarding classes: they are the same as structs but members are private by default.

I am still not a fan of it.
structs are far better for data, which most of it is and the few function pointers we need to deal with can be handled elsewhere.

I'll add it regardless